### ISSUE TITLE
Router bugs for orphaned articles

### ISSUE THREAD

**Issue by wilsonge** [joomla/joomla-cms#13978]
This is a writeup for reproduction of the issue I was initially trying to fix in #13747 but unfortunately it didn't work. So I'm opening it as an open issue

### Steps to reproduce the issue
Install staging (or latest version of Joomla 3.7.0 - currently beta 2) with no sample data.

In the article manager create 3 articles

1. Article 1 should be featured and uncategorised
2. Article 2 should not be featured and also uncategorised
3. Article 3 should be in a secondary category
4. Create a child menu item of the featured home page for article 1
5. Create a category list menu item for the secondary category as a top level menu item
6. Ensure that articles 1-3 all have links to each other. Note importantly here article 2 has no menu item and it's category has no menu item. This is running through lots of fallback behaviour and where bugs start to creep in.

Ensure you've turned on modern URL routing and Remove ID's for com_content

Firstly go to your single item menu item for article

### How to reproduce the bugs
Try navigating to article 2
Note that the article cannot be accessed. (EDIT: WITH #14031 this is now fixed)
Go to the category list view you create and go into article 3
Go to article 2
Note you go back to to the category 2 view and not article 2.

### System information (as much as possible)
Staging Joomla running on mac osx setup described here https://getgrav.org/blog/macos-sierra-apache-multiple-php-versions

### Additional comments
Router is a bit broken :(

**Comment by bayareajenn** [comment]
Duplicated problem exactly. 

Because Article One was a Featured Article, I could get to it from the "home" page - but it gave me a funky URL (/?view=article&id=53:article-2). 

Windows 7, FF, Rochen Host, PHP 5.6.

**Comment by thndrbck** [comment]
Followed instructions above.
Problem duplicated.
Cannot navigate to article 2 through the links when IDs are removed.

Ubuntu Linux with Chrome browser.
1&1 shared hosting.  subdomain.domain.tld
php7

clicking link to article 2 from the subcategory page (article 3) goes up one category level in the article 3 tree. Not to the uncategorized article 2.

When article 1 is in feature list, link to article 2 goes to article 2 but the title is for the original article. As if it is featured article list with article 2 subbed in as article 1. This behavior needs more exerimentation - more featured articles.

When article 1 is a menu item, link to article 2 goes back to self

**Comment by infograf768** [comment]
Indeed, badly broken, even if we create a hidden menu item for category list Uncategorised.
@Hackwar ...

**Comment by infograf768** [comment]
It works fine, happily..., when using legacy router.

**Comment by bayareajenn** [comment]
I decided to try out adding the hidden menu item for Article Two.  

When I added a Single Article menu item type for Article Two and then tried the links from Articles One & Three, it did work. I could get to Article Two nicely and the URL was /article-two from all the articles with links to it.

Side note: Interestingly, when I go to add a single article menu item type and then select my Article Two, my new articles don't appear in the options. I have two pages of articles and it was on NEITHER of them until I went BACK to page one again and then they display. This happened to me last night too (PST, in California) and I thought I was losing my mind but now that it's morning I know I'm not.

This also happens in the Article (Manager). My three new articles (Article 1, Article 2, and Article 3) are no where to be seen until I go to page two and then back to page one. Is that a different issue? I'm not very adept in github so I'll need direction if this is a new issue.

**Comment by bayareajenn** [comment]
I did the category list for Uncategorised menu item type next and it is working for me. (I unpublished my single article menu item for Article Two.)

When I click on Article Two in my content it takes me to a URL /cat-list-uncategorised/article-2 and I get there nicely.

**Comment by crleathers** [comment]
Duplicated the issue as described. Clicking link to article 2 from the subcategory page (article 3) goes up one category level in the article 3 tree, displaying a category listing page instead of "article 2".

MAMP Pro 3.5 on Mac Os Sierra running PHP 5.5.26
Firefox and Chrome and Safari on Mac<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13978">issues.joomla.org/tracker/joomla-cms/13978</a>.</sub>

**Comment by hypeitinc** [comment]
Followed instructions above.
Duplicated problem exactly.

**Also noticed the following:**
(the two links for the articles are created in two different ways)

Article 1 content: 
<p>Test article 1</p>
<a href=“.../article-2">Article 2</a>
<a href="index.php?option=com_content&view=article&id=127&catid=2">Test article 2</a>
<a href=“.../article-3">Article 3</a>
<a href="index.php?option=com_content&view=article&id=128&catid=10">Test article 3</a>

**Result:**
When hovering over the first link to Article 2 you get what you expect: .../article-2
When hovering over the second Article 2 link the link is incorrect, says: …/artists/article1
(article id 127 is in fact Article 2)
Clicking the first one gets you to the URL: /article-2  but the article does not display
Clicking the second one to:  /articles/article1

When hovering the over the first link to Article 3 you get what you expect: .../article-3
When hovering over the second Article 3 link the link is incorrect, says: .../artists/128-test-article-3
(article id 128 is in fact Article 3)
Clicking the first one gets you to the URL: /article-3  but the article does not display
Clicking the second one to:  /artists/128-test-article-3 but what displays is the first Manu Menu item!

**System information**
MacOS 10.11.6, Safari 10.0.3, Linux server, Apache, PHP 5.5.16, Joomla! 3.7.0-beta2 Beta [ Amani ] 8-February-2017 14:11 GMT

**Comment by Hackwar** [comment]
I think I found a solution. See #14031

**Comment by wilsonge** [comment]
@jeckodevelopment leaving this open as per my comment in #14031 as that only solved half of this issue

**Comment by Hackwar** [comment]
@hypeitinc I don't see the problem that you have. This is not really a bug in Joomla, but you misinterpreting the system. Joomla can only build from non-SEF links in content (simply because there is no sane way to find out if any other link actually points to Joomla or not) and when you invent your own URLs, it is no wonder that they don't work. The other issue has (partially) been solved.

**Comment by hypeitinc** [comment]
@Hackwar - Yes - I made a mistake there. However with the new router you don't get a 404 Not found. Anyway that's what I would expect to see. And, that's what I'm getting on a 3.6.5. site.
Also, when you hover over a link shouldn't it display the link with numbers stripped out?

**Comment by Hackwar** [comment]
@hypeitinc  

> Also, when you hover over a link shouldn't it display the link with numbers stripped out?

No.

**Comment by infograf768** [comment]
see also https://github.com/joomla/joomla-cms/pull/14322#issuecomment-284163611

**Comment by infograf768** [comment]
Better posted fully here
@Hackwar @rdeutz @wilsonge 
I found out while testing that
I needed to create an extraneous ROOT category for modern router to kind of work, making all other categories as children.
I.e. I could not get correct urls when creating only categories and a hidden menu item listing all categories from ROOT.
![screen shot 2017-03-04 at 16 34 33](https://cloud.githubusercontent.com/assets/869724/23579944/a9328828-00f8-11e7-944c-c2ed6c14ac28.png)

![screen shot 2017-03-04 at 16 33 18](https://cloud.githubusercontent.com/assets/869724/23579935/518d61ec-00f8-11e7-9a5c-99aae26aa1b2.png)

If I was using the list all categories based on ROOT, then it was impossible to get the articles.

![screen shot 2017-03-04 at 17 44 00](https://cloud.githubusercontent.com/assets/869724/23580436/3e59f8a6-0102-11e7-8b77-c71293b0f0eb.png)

Test:
Home page is a Search menu item.
Other menu item: List all articles categories.
Also used a Latest Articles Module
